### PR TITLE
Backwards compatibility for rstrnt-report-log

### PR DESCRIPTION
**Pull request by pfdaly**
This is the second command delivery to fulfil TMT #1027

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

File: tmt/steps/execute/scripts/tmt-file-submit [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+
+if [ $# -eq 2 ]; then
+    FILENAME=$2
+elif [ $# -gt 2 ]; then
+    check_opt_args "$@"
+fi
 [ -d "$TMT_TEST_DATA" ] || mkdir -p "$TMT_TEST_DATA"
-cp "$FILENAME" "$TMT_TEST_DATA"
+cp -f "$FILENAME" "$TMT_TEST_DATA"
 echo "File $FILENAME stored to $TMT_TEST_DATA"
```
Comment (psss) [inline tmt/steps/execute/scripts/tmt-file-submit:32]:
```suggestion
echo "File '$FILENAME' stored to '$TMT_TEST_DATA'."
```

File: stories/features/report-log.fmf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+title: Save a log file.
```
Comment (psss) [inline stories/features/report-log.fmf:1]:
```suggestion
title: Save a log file
```

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for writing this! Looks good, added just a few minor suggestions plus I've notice that upstream `restraint` documentation mentions `-l` as the parameter for providing the log while the patch uses `-p`.

File: stories/features/report-log.fmf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    guest and overwrite any existing scripts with the same name.
+
+    The command can be called multiple times for a single test,
+    if a log of that name already exists then it will be
+    overwritten.
+
+    __ https://restraint.readthedocs.io/en/latest/commands.html#rstrnt-report-log
+
+example: |
+        tmt-report-log -p /path/to/log/log.txt
```
[the remark this answers is not in the document]
Comment (pfdaly) [reply stories/features/report-log.fmf]: Indeed. I've corrected it in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: tests/execute/restraint/report-log/data/log.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+TMP_FILE="test_log_for_submitting.txt"
+TMP_FILE_SRC_PATH="/tmp/$TMP_FILE"
+TMP_FILE_DEST_PATH="$TMT_TEST_DATA/$TMP_FILE"
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "touch $TMP_FILE_SRC_PATH" 0 "Test log file successfully created."
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Verify mocked Restraint rstrnt-report-result file generated correctly.'
```
[the remark this answers is not in the document]
Comment (pfdaly) [reply tests/execute/restraint/report-log/data/log.sh]: It was. I've updated it to reflect the rstrnt-report-log command in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: tests/execute/restraint/report-log/test.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "pushd data"
+    rlPhaseEnd
+
+    rlPhaseStartTest
+        rlRun -s "tmt run -vvv --remove"
+        rlAssertGrep "pass /log" $rlRun_LOG
```
[the remark this answers is not in the document]
Comment (pfdaly) [reply tests/execute/restraint/report-log/test.sh]: That's true. I've removed in grep in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: tmt/steps/execute/scripts/tmt-file-submit [hunk 6]
```diff
@@ -1,5 +1,32 @@
...
+
+check_opt_args () {
+    local OPTIND
+    while getopts l:filename:s:server:port-: OPT; do
+        # support long options: https://stackoverflow.com/a/28466267/519360
+        if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
+            OPT="${OPTARG%%=*}"       # extract long option name
+        fi
+        case "$OPT" in
+            filename )         needs_arg; eval "FILENAME=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
```
[the remark this answers is not in the document]
Comment (pfdaly) [reply tmt/steps/execute/scripts/tmt-file-submit]: Updated in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: tmt/steps/execute/scripts/tmt-file-submit (hunk 1 above)
Comment (pfdaly) [reply tmt/steps/execute/scripts/tmt-file-submit]: Added single quotes and full stop in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: stories/features/report-log.fmf (hunk 2 above)
Comment (pfdaly) [reply stories/features/report-log.fmf]: Removed full stop character in commit [[249c948](https://github.com/teemtee/tmt/pull/1342/commits/249c948bfd797ece4b43eb89de5a4d033f428dca)]

File: tests/execute/restraint/report-log/data/log.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+TMP_FILE_SRC_PATH="/tmp/$TMP_FILE"
+TMP_FILE_DEST_PATH="$TMT_TEST_DATA/$TMP_FILE"
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "touch $TMP_FILE_SRC_PATH" 0 "Test log file successfully created."
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Verify mocked Restraint rstrnt-report-log correctly saves log file.'
+        rlRun "tmt-file-submit -p $TMP_FILE_SRC_PATH" 0 "Saving log with tmt-file-submit."
```
Comment (psss) [inline tests/execute/restraint/report-log/data/log.sh:14]: We need to fix the option in the test code as well.

**Review by psss — APPROVED** [review]
Thanks for quickly fixing all the issues! I've just added a reference to the `execute` step specification in 0e14294.

**Pull request merged**